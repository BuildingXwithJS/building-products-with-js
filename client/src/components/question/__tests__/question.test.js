// npm packages
import configureMockStore from 'redux-mock-store';

// our packages
import QuestionWrapper, {Question} from '../index';

// create mock store
const mockStore = configureMockStore();

const user = {
  id: '0',
  login: 'test',
};
const question = {
  owner: user,
  text: 'Question text',
  answers: [{answer: 'Test answer'}],
};

test('# QuestionWrapper', () => {
  const store = mockStore({auth: {user}});
  const wrapper = shallow(<QuestionWrapper question={question} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# Question', () => {
  const answer = 'Test answer';
  const updatedText = 'Updated text';
  const onAnswer = ({question: q, answer: a}) => {
    expect(q).toEqual(question);
    expect(a).toBe(answer);
  };
  const deleteQuestion = q => expect(q).toEqual(question);
  const updateQuestion = ({text}) => expect(text).toBe(updatedText);
  const component = (
    <Question
      user={user}
      question={question}
      onAnswer={onAnswer}
      deleteQuestion={deleteQuestion}
      updateQuestion={updateQuestion}
    />
  );
  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();
  // test interaction
  const app = mount(component);

  // test answer
  // set answer
  app.find('#answerInput').getDOMNode().value = answer;
  // click answer button
  app.find('#answerBtn').simulate('click');

  // test delete
  app.find('#deleteBtn').simulate('click');

  // test update
  // enable editing
  app.find('#editBtn').simulate('click');
  // change question text
  app.find('#questionText').getDOMNode().value = updatedText;
  // trigger update
  app.find('#updateBtn').simulate('click');
});
