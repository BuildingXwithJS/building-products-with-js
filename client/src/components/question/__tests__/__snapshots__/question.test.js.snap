// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Question 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <span>
      <button
        className="btn btn-link"
        id="deleteBtn"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-trash"
        />
      </button>
      <button
        className="btn btn-link"
        id="editBtn"
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-pencil"
        />
      </button>
    </span>
    Question text
    <div
      className="pull-right"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/profile/0"
      >
        test
      </Link>
    </div>
  </div>
  <div
    className="panel-body"
  >
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        Test answer
      </li>
    </ul>
  </div>
  <div
    className="panel-footer"
  >
    <form
      className="form-horizontal"
    >
      <div
        className="col-sm-10"
      >
        <input
          className="form-control"
          id="answerInput"
          placeholder="Enter your answer..."
          type="text"
        />
      </div>
      <button
        className="btn btn-default"
        id="answerBtn"
        onClick={[Function]}
        type="submit"
      >
        Answer
      </button>
    </form>
  </div>
</div>
`;

exports[`# QuestionWrapper 1`] = `
<Question
  deleteQuestion={[Function]}
  question={
    Object {
      "answers": Array [
        Object {
          "answer": "Test answer",
        },
      ],
      "owner": Object {
        "id": "0",
        "login": "test",
      },
      "text": "Question text",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  updateQuestion={[Function]}
  user={
    Object {
      "id": "0",
      "login": "test",
    }
  }
/>
`;
